docs/Secure-Coding-Guide-for-Python/CWE-664/CWE-665/README.md

```diff
@@ -1,10 +1,425 @@
 # CWE-665: Improper Initialization
 
+Prevent unexpected states by applying correct initialized of local objects as they remain available when a thread's resources are re-used in a thread-pool.
+
 > [!NOTE]
 > Prerequisite to understand this page:
 > [Intro to multiprocessing and multithreading](../../Intro_to_multiprocessing_and_multithreading/readme.md)
 
-## Introduction
+The `TheadPoolExecutor` provides an interface to a thread pool and resources which it reserved from the Operating System (OS). A thread in a thread-pool also known as "workers" can share an instance of a local object without overwriting each other's values inside of it [[Python docs 2023]](https://docs.python.org/3/library/threading.html#thread-local-data).
+
+> Thread-local data is data whose values are thread specific. To manage
+> thread-local data, just create an instance of local (or a subclass) and store attributes on it:
+> `mydata = threading.local()`
+> `mydata.x = 1`
+> The instance’s values will be different for separate threads.
+
+## Non-Compliant Code Example
+
+The `noncompliant01.py` code is simulating the spawning of processes with the different access rights.
+The `Session(object)` class simulates access control via `User(Enum)thread` defining a `User` object to be either `GUEST = 1` (default) or `ADMIN = 2`. The `set_user()` method  changes access level and `set_user_as_guest()` is to reset access.
+
+In `SessionPool()` we spawn tasks and access control in following order:

```

Comment:
This should probably say "in the following order" rather than "in following order"